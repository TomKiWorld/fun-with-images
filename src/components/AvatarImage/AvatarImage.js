import React from 'react';
import avatarOne from '../../images/FaceOne.svg';
import avatarTwo from '../../images/FaceTwo.svg';
import avatarThree from '../../images/FaceThree.svg';
import avatarFour from '../../images/FaceFour.svg';

const getAvatar = (avatarId) => {
  switch(avatarId) {
    case 'avatarOne':
      return avatarOne;
    case 'avatarTwo':
      return avatarTwo;
    case 'avatarThree':
      return avatarThree;
    case 'avatarFour':
      return avatarFour;
    default:
      return avatarOne;
  }      
}

const AvatarImage = ({ avatarId }) => {
  return (
    <img
      src={getAvatar(avatarId)}
      className='avatar-img' alt='avatar'
      />
  );
}

export default AvatarImage;
