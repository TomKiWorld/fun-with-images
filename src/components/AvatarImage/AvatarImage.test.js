import { shallow } from 'enzyme';
import React from 'react';
import AvatarImage from './AvatarImage';

describe('AvatarImage Component', () => {  
  it('Expect to render ImageCard Component', () => {
    expect(shallow(<AvatarImage />)).toMatchSnapshot();
  });

  it('Expect to render ImageCard Component with image for avatarOne', () => {
    const wrapper = shallow(<AvatarImage avatarId={'avatarOne'}/>)
    const image = wrapper.find('.avatar-img').at(0);
    expect(image.prop('src')).toBe('FaceOne.svg');
  });

  it('Expect to render ImageCard Component with image for avatarTwo', () => {
    const wrapper = shallow(<AvatarImage avatarId={'avatarTwo'}/>)
    const image = wrapper.find('.avatar-img').at(0);
    expect(image.prop('src')).toBe('FaceTwo.svg');
  });

  it('Expect to render ImageCard Component with image for avatarThree', () => {
    const wrapper = shallow(<AvatarImage avatarId={'avatarThree'}/>)
    const image = wrapper.find('.avatar-img').at(0);
    expect(image.prop('src')).toBe('FaceThree.svg');
  });

  it('Expect to render ImageCard Component with image for avatarFour', () => {
    const wrapper = shallow(<AvatarImage avatarId={'avatarFour'}/>)
    const image = wrapper.find('.avatar-img').at(0);
    expect(image.prop('src')).toBe('FaceFour.svg');
  });
});
