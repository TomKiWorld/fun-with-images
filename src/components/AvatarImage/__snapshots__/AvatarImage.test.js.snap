// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarImage Component Expect to render ImageCard Component 1`] = `ShallowWrapper {}`;
